tests/meta/demultiplex/test_demux_post_processing.py

```diff
@@ -167,52 +150,63 @@ def test_get_all_demultiplexed_flow_cell_out_dirs(
     assert tmp_demultiplexed_runs_bcl2fastq_directory in demultiplexed_flow_cell_dirs
 
 
-def test_post_processing_tracks_undetermined_fastqs_for_bclconvert(
-    demux_post_processing_api: DemuxPostProcessingAPI,
-    bclconvert_flow_cell_dir_name: str,
-    bcl_convert_sample_id_with_non_pooled_undetermined_reads: str,
-    bcl_convert_non_pooled_sample_read_count: int,
+def test_post_processing_tracks_undetermined_fastq_files(
+    updated_demux_post_processing_api: DemuxPostProcessingAPI,
+    hiseq_x_single_index_flow_cell: FlowCellDirectoryData,
+    selected_hiseq_x_single_index_sample_ids: list[str],
 ):
     # GIVEN a flow cell with undetermined fastqs in a non-pooled lane
 
+    # GIVEN that the flow cell has the sample sheet in housekeeper
+    add_and_include_sample_sheet_path_to_housekeeper(
+        flow_cell_directory=hiseq_x_single_index_flow_cell.path,
+        flow_cell_name=hiseq_x_single_index_flow_cell.id,
+        hk_api=updated_demux_post_processing_api.hk_api,
+    )
```

Comment:
was this step already added in the previous demux_post_processing_api? Or why has this behaviour changed?

Reply:
The fixture previously used (`low_cell_directory_name_demultiplexed_with_bcl_convert_flat`) added the sample sheet to HK in its definition. This one does not. I think it is debatable if the fixture of a file should include it to HK right away, but this is out of scope